packages/connector-tencent-sms/src/http.ts

```diff
@@ -0,0 +1,109 @@
+import type { BinaryToTextEncoding } from 'crypto';
+import crypto from 'crypto';
+
+import got from 'got';
+
+import type { TencentErrorResponse, TencentSuccessResponse } from '@/schema';
+import { tencentErrorResponse } from '@/schema';
+
+const endpoint = 'sms.tencentcloudapi.com';
+
+function sha256Hmac(message: string, secret: string): string;
+function sha256Hmac(message: string, secret: string, encoding: BinaryToTextEncoding): Buffer;
+
+function sha256Hmac(message: string, secret: string, encoding?: BinaryToTextEncoding) {
+  const hmac = crypto.createHmac('sha256', secret);
+
+  return encoding ? hmac.update(message).digest(encoding) : hmac.update(message).digest();
+}
+
+function getHash(message: string, encoding: BinaryToTextEncoding = 'hex') {
+  const hash = crypto.createHash('sha256');
+
+  return hash.update(message).digest(encoding);
+}
+
+function getDate(timestamp: number) {
+  const date = new Date(timestamp * 1000);
+  const year = date.getUTCFullYear();
+  const month = `0${date.getUTCMonth() + 1}`.slice(-2);
+  const day = `0${date.getUTCDate()}`.slice(-2);
+
+  return `${year}-${month}-${day}`;
+}
+
+export function isSmsErrorResponse(response: unknown): response is TencentErrorResponse {
+  const result = tencentErrorResponse.safeParse(response);
+
+  return result.success;
+}
+
+export function sendSMSRequest(
+  templateId: string,
+  templateParameters: string[],
+  phoneNumber: string,
+  config: {
+    secretId: string;
+    secretKey: string;
+    region: string;
+    sdkAppId: string;
+    signName: string;
+  }
+) {
+  const { secretId, secretKey, region, sdkAppId, signName } = config;
+  const timestamp = Math.floor(Date.now() / 1000);
```

Comment:
Why do we divide the timestamp by 1000, and then times 1000 again in the `getDate()` function?

Reply:
This is required for Tencent Cloud V3 signatures.